### PR TITLE
[TTS] Fix aligner nan loss in fp32

### PR DESCRIPTION
**Pull request by hsiehjackson**
# What does this PR do ?

Fix FastPitch training appear `nan` loss in aligner loss when precision=32.  

**Collection**: [TTS]

# Changelog 
- `nemo/collections/tts/losses/aligner_loss`

# Usage
* To reproduce the bug, we can run `FastPitch and Mixer-TTS Training` tutorial by adding `trainer.precision=32` in training command. 

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/tts/losses/aligner_loss.py [hunk 1]
```diff
@@ -58,9 +58,7 @@ def forward(self, attn_logprob, in_lens, out_lens):
         # Convert to log probabilities
         # Note: Mask out probs beyond key_len
         key_inds = torch.arange(max_key_len + 1, device=attn_logprob.device, dtype=torch.long)
-        attn_logprob.masked_fill_(
-            key_inds.view(1, 1, -1) > key_lens.view(1, -1, 1), -float("inf")  # key_inds >= key_lens+1
-        )
+        attn_logprob.masked_fill_(key_inds.view(1, 1, -1) > key_lens.view(1, -1, 1), -1e15)  # key_inds >= key_lens+1
         attn_logprob = self.log_softmax(attn_logprob)
```
Comment (rlangman) [inline nemo/collections/tts/losses/aligner_loss.py:62]: Should we make the masking logit value a parameter to the class, similar to "bank_logit"?

Comment (rlangman) [reply nemo/collections/tts/losses/aligner_loss.py:62]: Also I am curious if anyone knows why this would produce NaN. Does it happen immediately or a long time into training?

Sounds like a pretty large bug, if it happens for any fp32 training run.

Comment (hsiehjackson) [reply nemo/collections/tts/losses/aligner_loss.py:62]: In `FastPitch and Mixer-TTS Training` tutorial, this would appear immediately. In my own FastPitch pre-training, this appear after several epochs. I am not sure how this happen.

**Review by rlangman — APPROVED** [review]
